plugins/out_stackdriver/stackdriver_helper.c

```diff
@@ -0,0 +1,69 @@
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2019-2020 The Fluent Bit Authors
+ *  Copyright (C) 2015-2018 Treasure Data Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+
+#include "stackdriver.h"
+
+int cmp_helper(msgpack_object key, const char* str, const int size) {
```

Comment:
1. This only works when `key` is msgpack_object_str - I think other cases won't work. Change the function name to make it clear about the constraint, and document the premise of using this function. 

2. You should check the msppack_object's type first so it you are accessing `via` correctly. 

3. `key` is confusing because within this function. Just call it object as you don't know what the msgpack_object passed in yet.

Reply:
Done